consoles/sshSerial.pm

```diff
@@ -0,0 +1,82 @@
+# Copyright © 2018-2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+#
+# Simple serial terminal over SSH
+
+package consoles::sshSerial;
+
+use strict;
+use warnings;
+
+use base 'consoles::console';
+
+use consoles::ssh_screen;
+
+sub new {
+    my ($class, $testapi_console, $args) = @_;
+
+    my $self = $class->SUPER::new($testapi_console, $args);
+
+    return $self;
+}
+
+sub screen { shift->{screen} }
+
+sub disable {
+    my ($self) = @_;
+
+    return unless $self->{ssh};
+    bmwqemu::diag("Closing SSH connection with " . $self->{ssh}->hostname);
+    $self->{ssh}->disconnect;
+    $self->{ssh} = $self->{screen} = undef;
+    return;
+}
+
+sub activate {
+    my ($self) = @_;
+    my $hostname = $self->{args}->{hostname} || die('we need a hostname to ssh to');
+    my $password = $self->{args}->{password} // $testapi::password;
+    my $username = $self->{args}->{user}     // 'root';
+
+    bmwqemu::diag(sprintf("Connecting SSH serial console for %s@%s",
+            $username, $hostname));
+
+    my $ssh = $self->backend->new_ssh_connection(
+        hostname => $hostname,
+        password => $password,
+        username => $username
+    );
+    my $chan = $ssh->channel();
+
+    $chan->blocking(0);
+
+    # Enable echo, no ANSI color codes, 2048 character line width
+    # (Sending commands longer than line width will break read-back check)
+    $chan->pty('dumb', {echo => 1}, 2048);
```

Comment:
This looks like a magic number that might be useful to make configurable. Or is this bound to ssh internals which we maybe can read out from there instead of hard coding here?

Reply:
Added `SSH_PTY_WIDTH` constant to configure this. It's not something internal to SSH, it's a setting needed to prevent the remote side from chopping off the beginning of echoed strings that are too long for the emulated terminal.